### DISCUSSION TITLE
Is it possible to disable GitHub-hosted runners?

### DISCUSSION THREAD

**Discussion by carlturnerfs** [community/community#39969] [Actions] [answered]
### Select Topic Area

Question

### Body

Is it possible to disable GitHub-hosted runners and force the use of self-hosted runners (regardless of what someone writes in the YAML configuration)?

I have tried setting the `Default` Runner group to an empty list of repositories but Actions still appear to happily execute on hosted runners regardless.

**Comment by ossanna16** [comment]
Hi there @carlturnerfs and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Comment by carlturnerfs** [comment, marked as answer]
The following is from GitHub Support:

> There is currently no setting for preventing workflows from running on GitHub-hosted runners.
> 
> Assuming that your goal is to restrict users to the use of self-hosted runners, a possible solution is to add a workflow to your repositories that is run whenever a push or pull request is modifying a file in the path .github/workflows/**. The purpose of the workflow would be to check the definition of the modified/added workflow file to ensure that all of the jobs.<job_id>.runs-on values use self-hosted runners instead of GitHub-hosted runners, rejecting the change if the workflow does not match the requirement.
> 
> You can restrict self-hosted runner groups to only be accessible from certain workflows see blog post.
> 
> While we don't currently have that functionality implemented, I've gone ahead and added your organization's voice to the internal Issue tracking the feature request and looped in your account representative. At this time we can't say when or if it will be released but we'd recommend that you keep an eye on the GitHub changelog where updates are announced.
